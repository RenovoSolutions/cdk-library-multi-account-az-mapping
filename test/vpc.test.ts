import {
  Stack,
  App,
  aws_ec2 as ec2,
  aws_ssm as ssm,
} from 'aws-cdk-lib';
import 'jest-cdk-snapshot';
import {
  AzIdToNameMapping,
} from '../src/index';

test('Snapshot', () => {
  const app = new App();
  const stack = new Stack(app, 'TestStack');

  const mapping = new AzIdToNameMapping(stack, 'AzIdToNameMapping', {});

  const vpc = new ec2.Vpc(stack, 'vpc', {
    cidr: '10.0.0.0/16',
    availabilityZones: [
      ssm.StringParameter.fromStringParameterName(stack, 'az1', '/az-mapping/az1').stringValue,
      ssm.StringParameter.fromStringParameterName(stack, 'az2', '/az-mapping/az2').stringValue,
    ],
  });

  vpc.node.addDependency(mapping);

  expect(stack).toMatchCdkSnapshot({
    ignoreAssets: true,
  });
});