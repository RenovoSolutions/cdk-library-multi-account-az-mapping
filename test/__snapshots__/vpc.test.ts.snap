// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Parameters": Any<Object>,
  "Resources": Object {
    "AzIdToNameMappinghandler3666E550": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerRole0E94DF36",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Description": "Stores VPC mappings into parameter store",
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "AzIdToNameMappinghandlerRole0E94DF36",
            "Arn",
          ],
        },
        "Runtime": "python3.13",
        "Timeout": 5,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AzIdToNameMappinghandlerLogGroupCDB14C15": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "AzIdToNameMappinghandler3666E550",
              },
            ],
          ],
        },
        "RetentionInDays": 731,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "AzIdToNameMappinghandlerPolicyBA4D5D94": Object {
      "Properties": Object {
        "Description": "",
        "Path": "/",
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ssm:PutParameter",
                "ssm:DeleteParameter",
                "ssm:AddTagsToResource",
              ],
              "Condition": Object {
                "StringEquals": Object {
                  "aws:ResourceTag/CdkAzMappingParameter": "True",
                },
              },
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "AllowSSMParameterEdit",
            },
            Object {
              "Action": "ec2:DescribeAvailabilityZones",
              "Effect": "Allow",
              "Resource": "*",
              "Sid": "AllowDescribeAZs",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::ManagedPolicy",
    },
    "AzIdToNameMappinghandlerRole0E94DF36": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Ref": "AzIdToNameMappinghandlerPolicyBA4D5D94",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "AzIdToNameMappinglogGroup3282D2C7": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "AzIdToNameMappinghandler3666E550",
              },
            ],
          ],
        },
        "RetentionInDays": 30,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "AzIdToNameMappingmappingFF3EB1A2": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "AzIdToNameMappinghandler3666E550",
            "Arn",
          ],
        },
        "azIds": Array [
          "use1-az2",
          "use1-az4",
          "use1-az6",
        ],
        "prefix": "/az-mapping/",
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "vpcA2121C38": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "vpcIGWE57CBDCA": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "vpcPrivateSubnet1DefaultRoute1AA8E2E5": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "vpcPublicSubnet1NATGateway9C16659E",
        },
        "RouteTableId": Object {
          "Ref": "vpcPrivateSubnet1RouteTableB41A48CC",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPrivateSubnet1RouteTableAssociation67945127": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcPrivateSubnet1RouteTableB41A48CC",
        },
        "SubnetId": Object {
          "Ref": "vpcPrivateSubnet1Subnet934893E8",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPrivateSubnet1RouteTableB41A48CC": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPrivateSubnet1Subnet934893E8": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "AvailabilityZone": Object {
          "Ref": "az1Parameter",
        },
        "CidrBlock": "10.0.128.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcPrivateSubnet2DefaultRouteB0E07F99": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "vpcPublicSubnet2NATGateway9B8AE11A",
        },
        "RouteTableId": Object {
          "Ref": "vpcPrivateSubnet2RouteTable7280F23E",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPrivateSubnet2RouteTable7280F23E": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPrivateSubnet2RouteTableAssociation007E94D3": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcPrivateSubnet2RouteTable7280F23E",
        },
        "SubnetId": Object {
          "Ref": "vpcPrivateSubnet2Subnet7031C2BA",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPrivateSubnet2Subnet7031C2BA": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "AvailabilityZone": Object {
          "Ref": "az2Parameter",
        },
        "CidrBlock": "10.0.192.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcPublicSubnet1DefaultRoute10708846": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
        "vpcVPCGW7984C166",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "vpcIGWE57CBDCA",
        },
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet1RouteTable48A2DF9B",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPublicSubnet1EIPDA49DCBE": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "vpcPublicSubnet1NATGateway9C16659E": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
        "vpcPublicSubnet1DefaultRoute10708846",
        "vpcPublicSubnet1RouteTableAssociation5D3F4579",
      ],
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "vpcPublicSubnet1EIPDA49DCBE",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "vpcPublicSubnet1Subnet2E65531E",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "vpcPublicSubnet1RouteTable48A2DF9B": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPublicSubnet1RouteTableAssociation5D3F4579": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet1RouteTable48A2DF9B",
        },
        "SubnetId": Object {
          "Ref": "vpcPublicSubnet1Subnet2E65531E",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPublicSubnet1Subnet2E65531E": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "AvailabilityZone": Object {
          "Ref": "az1Parameter",
        },
        "CidrBlock": "10.0.0.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcPublicSubnet2DefaultRouteA1EC0F60": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
        "vpcVPCGW7984C166",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "vpcIGWE57CBDCA",
        },
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet2RouteTableEB40D4CB",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcPublicSubnet2EIP9B3743B1": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "vpcPublicSubnet2NATGateway9B8AE11A": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
        "vpcPublicSubnet2DefaultRouteA1EC0F60",
        "vpcPublicSubnet2RouteTableAssociation21F81B59",
      ],
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "vpcPublicSubnet2EIP9B3743B1",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "vpcPublicSubnet2Subnet009B674F",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "vpcPublicSubnet2RouteTableAssociation21F81B59": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "vpcPublicSubnet2RouteTableEB40D4CB",
        },
        "SubnetId": Object {
          "Ref": "vpcPublicSubnet2Subnet009B674F",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcPublicSubnet2RouteTableEB40D4CB": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcPublicSubnet2Subnet009B674F": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "AvailabilityZone": Object {
          "Ref": "az2Parameter",
        },
        "CidrBlock": "10.0.64.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcVPCGW7984C166": Object {
      "DependsOn": Array [
        "AzIdToNameMappinghandlerLogGroupCDB14C15",
        "AzIdToNameMappinghandler3666E550",
        "AzIdToNameMappinghandlerPolicyBA4D5D94",
        "AzIdToNameMappinghandlerRole0E94DF36",
        "AzIdToNameMappinglogGroup3282D2C7",
        "AzIdToNameMappingmappingFF3EB1A2",
      ],
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "vpcIGWE57CBDCA",
        },
        "VpcId": Object {
          "Ref": "vpcA2121C38",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
}
`;
